bootstrap.sh

```diff
@@ -1,112 +1,142 @@
+
+		# source is done and tagged
+		cd ..
+		touch tagged-src
+		cd ..
+	fi
+}
+
+bootstrap_columns() {
+	echo "=== Bootstrapping columns ==="
+
+	mkdir -p build/stage1/bin
+	cd build/src
+	git clean -fdx
+	git restore --staged --worktree :/
+	patch -p1 -i ../../columns.patch
+	cd columns
+	"${CC:-gcc}" ${CFLAGS} columns.c -o "${PREFIX}/bin/columns"
+	cd ../../..
+}
+
+bootstrap_getdefs() {
+	echo "=== Bootstrapping getdefs ==="
+
+	cd build/src
+	git clean -fdx
+	git restore --staged --worktree :/
+	patch -p1 -i ../../getdefs.patch
+	cd getdefs
+	"${CC:-gcc}" ${CFLAGS} -std=gnu99 getdefs.c -I ../../.. -o "${PREFIX}/bin/getdefs"
+	cd ../../..
+}
+
+bootstrap_autogen() {
+	echo "=== Bootstrapping autogen ==="
+
+	cd build/src
+	git clean -fdx
+	git restore --staged --worktree :/
+	patch -p1 -i ../../autogen.patch
+	cd snprintfv
+	bash bootstrap.dir
+	cd ../agen5
+	perl ../../../build-ag-text.pl
+	"${PREFIX}/bin/getdefs" output=functions.def template=functions.tpl srcfile linenum defs=macfunc listattr=alias $(grep -l '/\*=macfunc' *.c)
+	"${PREFIX}/bin/getdefs" load=expr.cfg
+	sed -n '/^doDir_invalid/d;/^doDir_[a-z]*(/{;s@(.*@@;s@^doDir_@@;p;}' defDirect.c | sort >directive_in.def
+	perl ../../../build-indirect-templates.pl
+	"${CC:-gcc}" ${CFLAGS} ../../../getGuileVersion.c $(pkg-config guile-"${GUILE_VERSION}" --cflags) -o getGuileVersion
+	"${CC:-gcc}" ${CFLAGS} -std=gnu99 -DGUILE_VERSION=$(./getGuileVersion) -DLIBDATADIR=\"$PREFIX/lib/autogen\" ../../../agBootstrap.c -I . -I .. -I ../../.. $(pkg-config guile-"${GUILE_VERSION}" --cflags "${GUILE_STATIC}") -o $PREFIX/bin/autogen $(pkg-config guile-"${GUILE_VERSION}" --libs "${GUILE_STATIC}")
+	cd ../../..
+}
+
+bootstrap_tpl_config() {
+	echo "=== Bootstrapping tpl-config.tlib ==="
+
+	mkdir -p build/stage1/lib/autogen
+	cd build/src
+	export PATH="$PREFIX/bin:$PATH"
+
+	git clean -fdx
+	git restore --staged --worktree :/
+	sed 's/@EGREP@/egrep/g;s/@GREP@/grep/g' <autoopts/tpl/tpl-config-tlib.in >"${PREFIX}/lib/autogen/tpl-config.tlib"
+	cp autoopts/tpl/*.lic "${PREFIX}/lib/autogen"
+	SOURCE_DIR="$(pwd)" ./config/bootstrap
+	./configure --prefix="$PREFIX" --disable-dependency-tracking ${CONFIGURE_FLAGS}
+	cd autoopts
+	make tpl-config-stamp
+	cp tpl/tpl-config.tlib "${PREFIX}/lib/autogen/tpl-config.tlib"
+	cd ../../..
+}
+
+rebuild_autogen() {
+	echo "=== Compiling and testing package ==="
+
+	cd build/src
+	git clean -fdx
+	git restore --staged --worktree :/
+	SOURCE_DIR="$(pwd)" ./config/bootstrap
+	./configure --prefix="$FINALPREFIX" --disable-dependency-tracking ${CONFIGURE_FLAGS}
+	touch doc/agdoc.texi # build later
+	make CFLAGS=-Wno-error
+	make check
+	make install
+	cd doc
+	PATH="$FINALPREFIX/bin:$PATH"
+	rm agdoc.texi
+	make
+	make install
+	cd ../../..
+
+	echo "=== BOOTSTRAP COMPLETE ==="
+}
+
+if [ ${SKIP_MAIN:-0} -ne 1 ]; then
```

Comment:
How about

```suggestion

if [ -n "${SKIP_MAIN}" ]; then
```

Reply:
Sure, I can do that. I'll apply it to the other file too.

Reply:
Actually this doesn't work as it skips main by default

Reply:
yeah, I meant `-z` and not `-n`, sorry. :(